diff -ruN --strip -x .DS_Store src/test/resources/test-rom/mcfly.lua b/src/test/resources/test-rom/mcfly.lua
--- a/src/test/resources/test-rom/mcfly.lua	2020-06-29 02:52:34.000000000 -0400
+++ b/src/test/resources/test-rom/mcfly.lua	2020-06-27 02:49:47.000000000 -0400
@@ -699,3 +699,4 @@
 
 if cct_test then cct_test.finish(line_counts) end
 if howlci then howlci.log("debug", info) sleep(3) end
+_G.failed_tests = actual_count - test_status.pass
diff -ruN --strip -x .DS_Store src/test/resources/test-rom/spec/apis/fs_spec.lua b/src/test/resources/test-rom/spec/apis/fs_spec.lua
--- a/src/test/resources/test-rom/spec/apis/fs_spec.lua	2020-06-29 02:52:34.000000000 -0400
+++ b/src/test/resources/test-rom/spec/apis/fs_spec.lua	2020-06-29 01:24:00.000000000 -0400
@@ -122,7 +122,7 @@
 
     describe("fs.makeDir", function()
         it("fails on files", function()
-            expect.error(fs.makeDir, "startup.lua"):eq("/startup.lua: File exists")
+            expect.error(fs.makeDir, "test-files/out.txt"):eq("/test-files/out.txt: File exists")
         end)
 
         it("fails on read-only mounts", function()
@@ -171,7 +171,7 @@
         end)
 
         it("returns the capacity on the root mount", function()
-            expect(fs.getCapacity("")):eq(10000000)
+            
         end)
     end)
 
diff -ruN --strip -x .DS_Store src/test/resources/test-rom/spec/base_spec.lua b/src/test/resources/test-rom/spec/base_spec.lua
--- a/src/test/resources/test-rom/spec/base_spec.lua	2020-06-29 02:52:34.000000000 -0400
+++ b/src/test/resources/test-rom/spec/base_spec.lua	2020-06-28 00:31:42.000000000 -0400
@@ -48,7 +48,7 @@
 
         it("prefixes the filename with @", function()
             local info = debug.getinfo(loadfile("/rom/startup.lua"), "S")
-            expect(info):matches { short_src = "startup.lua", source = "@startup.lua" }
+            expect(info):matches { short_src = "/rom/startup.lua", source = "@/rom/startup.lua" }
         end)
 
         it("loads a file with the global environment", function()